src/marqo/core/semi_structured_vespa_index/semi_structured_vespa_schema_template.sd.jinja2

```diff
@@ -0,0 +1,253 @@
+            attribute {
+                distance-metric: {{ index.distance_metric.value }}
+            }
+            index {
+                hnsw {
+                    max-links-per-node: {{ index.hnsw_config.m }}
+                    neighbors-to-explore-at-insert: {{ index.hnsw_config.ef_construction }}
+                }
+            }
+        }
+        {% endfor -%}
+
+        field marqo__vector_count type int {
+            indexing: attribute | summary
+        }
+    }
+
+    {% if index.lexical_field_map -%}
+    fieldset default {
+        fields: {{ ", ".join(index.lexical_field_map.keys()) }}
+    }
+    {% endif -%}
+
+    rank-profile base_rank_profile inherits default {
+        inputs {
+            {% for lexical_field in index.lexical_fields -%}
+            query({{ lexical_field.lexical_field_name }}): 0
+            {% endfor -%}
+            {% for field in index.tensor_fields -%}
+            query({{ field.embeddings_field_name }}): 0
+            {% endfor -%}
+            query(marqo__bm25_aggregator): 0
+            query(marqo__query_embedding) tensor<float>(x[{{ dimension }}])
+            query(marqo__mult_weights_lexical) tensor<double>(p{})
+            query(marqo__add_weights_lexical) tensor<double>(p{})
+            query(marqo__mult_weights_tensor) tensor<double>(p{})
+            query(marqo__add_weights_tensor) tensor<double>(p{})
+        }
+
+        function modify(score, mult_weights, add_weights) {
+            expression: if (count(mult_weights * attribute(marqo__score_modifiers)) == 0,   1, reduce(mult_weights * attribute(marqo__score_modifiers), prod)) * score + reduce(add_weights * attribute(marqo__score_modifiers), sum)
+        }
+
+        {% macro max_score(score_macro, fields) -%}
+            {%- set length = fields|length -%}
+            {%- if length == 0 -%}
+                0
+            {%- elif length == 1 -%}
+                {{ score_macro(fields[0]) }}
+            {%- else -%}
+                max({{ max_score(score_macro, fields[0:1]) }}, {{ max_score(score_macro, fields[1:]) }})
+            {%- endif -%}
+        {% endmacro -%}
+
+        {% macro lexical_score(field) -%}
+            if(query({{ field.lexical_field_name }}) > 0, bm25({{ field.lexical_field_name }}), 0)
+        {%- endmacro -%}
+
+        {% macro embedding_score(field) -%}
+            if(query({{ field.embeddings_field_name }}) > 0, closeness(field, {{ field.embeddings_field_name }}), 0)
+        {%- endmacro -%}
+
+        {% macro sum_score(score_macro, fields) -%}
+            {%- set add = joiner(" + ") -%}
+            {%- for field in fields -%}{{ add() }}{{ score_macro(field) }}{%- endfor -%}
+        {% endmacro -%}
+
+        {% if index.lexical_fields -%}
+        function lexical_score_sum() {
+            expression: {{ sum_score(lexical_score, index.lexical_fields) }}
+        }
+
+        function lexical_score_avg() {
+            {# TODO calculate average lexical score -#}
```

Comment:
@farshidz , Joshua and I had a discussion regarding this function for structured index here: https://marqo-ai.slack.com/archives/C05K6BFQUTF/p1726537136109449?thread_ts=1726530564.502089&cid=C05K6BFQUTF . 
Could you shed some lights on why we calculate average in this way? Thanks. 
BTW, I just leave it as 0 for now since we only use lexical_score_sum() right now.

Reply:
This seems like new code. Looking at existing tests the average function looks valid?